### ISSUE TITLE
Implement the Basic One-Shot Algorithm

### ISSUE THREAD

**Issue by emiliolr** [ceh-2000/gpu_guru#4]
Implement the one-shot FL algorithm described in [Guha et al. (2019)](https://arxiv.org/abs/1902.11175)--just the ensembling version that doesn't require auxiliary data.

- [x] Add a command argument to specify the sampling method for users. Add methods in the extended server class that allow for sampling via validation, amount of data, random sampling, and all users.
- [x] Implement an extended server class that overwrites the `create_users` method in the base class to split data subsets into additional training/validation subsets (if necessary for the sampling scheme).
- [x] The `train` method should allow all users to train and then should have selected users upload their models.
- [x] The evaluation step for this algorithm (the "global model") should just be based on the ensembled predictions (majority vote over classes or average logits) of selected/uploaded user models.
